### PR TITLE
Domsdatabasen API

### LINKED ISSUE #2: Support data from domsdatabasen
This should be a wrapper around the [existing domsdatabasen dataset on Hugging Face](https://huggingface.co/datasets/oliverkinch/domsdatabasen).

It should allow users to query entries, and these should use cached versions from the HF dataset, and otherwise scrape and parse these directly, using [this repo, which should be set up as a Python package](https://github.com/alexandrainst/doms_databasen).

### PR DESCRIPTION
**Pull request by oliverkinch**
Example usage

```
from alexandra_ai_data.domsdatabasen import Domsdatabasen

domsdatabasen = Domsdatabasen()

case = domsdatabasen.get_case(case_id="100")

print(case)

```

Fix https://github.com/alexandrainst/alexandra_ai_data/issues/2

### CODE REVIEW

File: src/alexandra_ai_data/domsdatabasen.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        dataset_builder (DatasetBuilder):
+            DatasetBuilder object for building dataset samples.
+        dataset (Dataset):
+            Dataset of processed data from Domsdatabasen.
+    """
+
+    def __init__(self):
+        """Initialize."""
+        self.config: DictConfig = config
+        self.dataset = load_dataset("alexandrainst/domsdatabasen")["train"]
```
Comment (saattrupdan) [inline src/alexandra_ai_data/domsdatabasen.py]: Consider using the `split` argument instead:

```suggestion
        self.dataset = load_dataset("alexandrainst/domsdatabasen", split="train")
```

File: src/alexandra_ai_data/domsdatabasen.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        Args:
+            case_id (str):
+                The case_id of the case to get data for.
+
+        Returns:
+            dataset_sample (dict):
+                Processed data for the case.
+        """
+        if isinstance(case_id, int):
+            case_id = str(case_id)
```
Comment (saattrupdan) [inline src/alexandra_ai_data/domsdatabasen.py:57]: If integer inputs are allowed, then this should be stated in the type signature.

File: src/alexandra_ai_data/domsdatabasen.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        self._initialze_objects()
+        self.scraper.scrape(case_id=case_id)
+        processed_data = self.processor.process(case_id=case_id)
+        dataset_sample = self.dataset_builder.make_dataset_sample(
+            processed_data=processed_data
+        )
+
+        return dataset_sample
+
+    def _initialze_objects(self):
```
Comment (saattrupdan) [inline src/alexandra_ai_data/domsdatabasen.py]: Typo:

```suggestion
    def _initialize_objects(self):
```

File: src/alexandra_ai_data/domsdatabasen.py [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+            if dataset_sample["case_id"] == case_id:
+                logger.info(f"Found case_id {case_id} in cached dataset.")
+                return dataset_sample
+
+        # If case_id is not in dataset, scrape and process the case
+        logger.info(
+            f"Case_id {case_id} not found in cached dataset. "
+            "Scraping and processing the case..."
+        )
+        self._initialze_objects()
```
Comment (saattrupdan) [inline src/alexandra_ai_data/domsdatabasen.py]: Typo:

```suggestion
        self._initialize_objects()
```

File: src/scripts/domsdatabasen_example_usage.py [hunk 5]
```diff
@@ -0,0 +1,9 @@
+"""Example usage of the Domsdatabasen API"""
+
+from alexandra_ai_data.domsdatabasen import Domsdatabasen
+
+domsdatabasen = Domsdatabasen()
+
+case = domsdatabasen.get_case(case_id="100")
+
+print(case)
```
Comment (saattrupdan) [inline src/scripts/domsdatabasen_example_usage.py]: Consider moving these kinds of examples to a separate `examples` directory, in the root folder.

Alternatively, you could simply add this bit of code to the readme under the Quick Start section.

**Review by saattrupdan — CHANGES_REQUESTED** [review]
Looks great! 🎉 

I've only got a few suggestions.

Also, I added a PR to merge in with your branch, which just updates the dependencies, Github workflows and such, since we haven't updated the repo in some time. No changes to your code in that PR, so it would be great if you could review that PR into this branch as well.

**Comment by oliverkinch** [comment]
I have merged your PR and made commits w.r.t your requested changes @saattrupdan

File: makefile [hunk 6]
```diff
@@ -41,7 +41,10 @@ install: ## Install dependencies
 	@$(MAKE) --quiet install-poetry
 	@$(MAKE) --quiet setup-poetry
 	@$(MAKE) --quiet setup-environment-variables
-	@echo "Installed the 'ScandEval' project."
```
Comment (saattrupdan) [inline makefile:44]: Whoops, I missed this bit. Could you rename "ScandEval" to "AlexandraAI-data" the two places in `make install`?

**Review by saattrupdan — APPROVED** [review]
Looks great! Just a small typo that was my mistake 🙂

File: makefile (hunk 6 above)
Comment (oliverkinch) [reply makefile:44]: Done [a87d4b5](https://github.com/alexandrainst/alexandra_ai_data/pull/5/commits/a87d4b55230545d3364c1f2ed5d0295856eceec7)

**Pull request merged**